### PR TITLE
Add allergies

### PR DESCRIPTION
**Pull request by BNAndras**
@exercism/ballerina, here's another one for #48in24. This one will be featured two weeks from now.

### CODE REVIEW

**Comment by BNAndras** [comment]
Seems like the CI doesn't like how I added the groups. The code compiled locally without issue, but I can remove the groups if that's the problem.

File: exercises/practice/allergies/.meta/reference/allergies.bal [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    function allergicTo(string candidate) returns boolean {
+        return self.list().some(allergen => allergen == candidate);
+    }
+
+    function list() returns string[] {
+        string[] results = [];
+        foreach [int, string] [i, allergen] in self.allergens.enumerate() {
+            if ((self.score & 1 << i) != 0) {
```
Comment (DimuthuMadushan) [inline exercises/practice/allergies/.meta/reference/allergies.bal:26]:
```suggestion
            if (self.score & 1 << i) != 0 {
```

File: exercises/practice/allergies/.meta/reference/allergies.bal [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        "pollen",
+        "cats"
+    ];
+
+    function init(int score) {
+        self.score = score;
+    }
+
+    function allergicTo(string candidate) returns boolean {
+        return self.list().some(allergen => allergen == candidate);
```
Comment (DimuthuMadushan) [inline exercises/practice/allergies/.meta/reference/allergies.bal:20]:
```suggestion
        return self.allergens.some(allergen => allergen == candidate);
```

**Comment by BNAndras** [comment]
CI passes so I'm ready for any further review. :)

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**